src/ipc/ipc4/helper.c

```diff
@@ -696,6 +696,11 @@ int ipc4_create_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 	}
 
 	ipc_pipe = ipc_get_comp_by_id(ipc, pipeline_id);
+	if (!ipc_pipe) {
```

Comment:
@jsarha this was already discussed and qualified as a false positive. The pipeline is created just above on line 691 and if it failed, we would never get here

Reply:
Maybe an assert could be good here though. Full validation? Probably not, but an assert to make it clear that you know it's good should be there. This isn't hot code anyway.